src/main.ts

```diff
@@ -1,6 +1,75 @@
 import { DwnServer } from './dwn-server.js';
+import { Dwn, DataStoreLevel, EventLogLevel, MessageStoreLevel } from '@tbd54566975/dwn-sdk-js';
+import { SqliteDialect, MysqlDialect, PostgresDialect, MessageStoreSql, DataStoreSql, EventLogSql } from '@tbd54566975/dwn-sql-store';
+
+import Database from 'better-sqlite3';
+import { createPool as MySQLCreatePool } from 'mysql2';
+import pg from 'pg';
+import Cursor from 'pg-cursor';
+
 
 const dwnServer = new DwnServer();
+
+// Initialize storage based on DWN_STORAGE environment variable
+const storageURLstring = process.env.DWN_STORAGE || 'level://data';
+const storageURL = new URL(storageURLstring);
+
+switch(storageURL.protocol) {
+case 'level:': {
+  const storageDir = storageURL.host + storageURL.pathname;
+  console.log('using leveldb storage in', storageDir);
+  const dataStore = new DataStoreLevel({ blockstoreLocation: storageDir + '/DATASTORE' });
+  const eventLog = new EventLogLevel({ location: storageDir + '/EVENTLOG' });
+  const messageStore = new MessageStoreLevel({
+    blockstoreLocation : storageDir + '/MESSAGESTORE',
+    indexLocation      : storageDir + '/INDEX'
+  });
+
+  dwnServer.dwn = await Dwn.create({ eventLog, dataStore, messageStore });
+  break;
+}
+
+case 'sqlite:': {
+  const db = new SqliteDialect({
+    database: async () => new Database(storageURL.host)
+  });
+
+  const messageStore = new MessageStoreSql(db);
+  const dataStore = new DataStoreSql(db);
+  const eventLog = new EventLogSql(db);
+  dwnServer.dwn = await Dwn.create({ eventLog, dataStore, messageStore });
+  break;
+}
+
+case 'mysql:': {
+  const db = new MysqlDialect({
+    pool: async () => MySQLCreatePool(storageURLstring),
+  });
+
+  const messageStore = new MessageStoreSql(db);
+  const dataStore = new DataStoreSql(db);
+  const eventLog = new EventLogSql(db);
+  dwnServer.dwn = await Dwn.create({ eventLog, dataStore, messageStore });
+  break;
+}
+
+case 'postgres:': {
```

Comment:
I'd suggest us relying on enums instead of string here, it's usually easier to maintain that way